lib/AST/ASTMangler.cpp

```diff
@@ -3237,8 +3226,9 @@ void ASTMangler::appendFunctionInputType(
           Identifier(), param.getPlainType(),
           getParameterFlagsForMangling(param.getParameterFlags(),
                                        defaultSpecifier, isRecursedInto),
-          lifetimeDependenceInfo.getLifetimeDependenceOnParam(paramIndex), sig,
-          nullptr);
+          getLifetimeDependenceFor(fnType->getLifetimeDependencies(),
```

Comment:
Do we need some deployment target checks here so that we don't run into the same back-deployment runtime crash as with isolated function types (https://github.com/swiftlang/swift/pull/74916)?
For example, if a struct contains a closure with the new lifetime dependencies, then an old runtime cannot demangle such a type.

Reply:
Looks like we definitely need that. Thanks for the pointer.

Reply:
Included this in the new change. No metadata test because we don't allow lifetime dependencies on function types yet.